### PR TITLE
Flash messages

### PR DESCRIPTION
**Pull request by spencerldixon**
Gordon's alive!

### CODE REVIEW

File: app/assets/javascripts/stores/questionnaire_store.cjsx [hunk 1]
```diff
@@ -195,11 +195,29 @@ class QuestionnaireStore extends EventEmitter
...
     $notificationEl = $('.js-questionnaire-notification')
-    $notificationEl.html(msg).fadeIn()
+    $notificationEl.append(
+      $('<div class="alert alert-' + key + '"></div>').append(
+        $('<p class="alert__close"><i class="fa fa-close"></i> Close</p>')
+      ).append(
+        $('<h4 class="alert__title"><i class="fa fa-' + title_icon + '"></i> ' + title + '</h4>')
+      ).append(
+        $('<p class="alert__body">' + msg + '</p>')
+        )
```
Comment (lucidstack) [inline app/assets/javascripts/stores/questionnaire_store.cjsx]: A little mess-up here with the indentation!

File: app/assets/javascripts/stores/questionnaire_store.cjsx [hunk 2]
```diff
@@ -195,11 +195,29 @@ class QuestionnaireStore extends EventEmitter
...
+      )
 
-    setTimeout((-> $notificationEl.fadeOut()), 3000)
+    $notificationEl.fadeIn()
+    setTimeout((-> $notificationEl.fadeOut()), 10000)
```
Comment (lucidstack) [inline app/assets/javascripts/stores/questionnaire_store.cjsx]: 10s seems a bit much! Did you mean 1s?

**Comment by lucidstack** [comment]
Just noticed that every time I save a report, new notifications keep appearing, instead of reusing the same block.

screen shot 2016-08-25 at 14 13 54

File: app/assets/javascripts/stores/questionnaire_store.cjsx (hunk 2 above)
Comment (rogeringle) [inline app/assets/javascripts/stores/questionnaire_store.cjsx]: @lucidstack We couldn't decide, 1s didn't seem long enough for longer messages. But 10s is probably too long.

Maybe we could change to something like 3s and then revisit later?

Comment (spencerldixon) [inline app/assets/javascripts/stores/questionnaire_store.cjsx]: Like the 3 seconds it was originally? 😉

Comment (lucidstack) [inline app/assets/javascripts/stores/questionnaire_store.cjsx]: 10 seconds is probably fine, actually. I added that comment before checking how the flash message is displayed. It has its own space in the page, it's not a fixed hovering thing, so 10s is probably fine

Comment (spencerldixon) [inline app/assets/javascripts/stores/questionnaire_store.cjsx]: You can click to dismiss it too, and we may change it to be a fixed hovering thing, but I want to style the questionnaire itself before we worry about that

**Pull request merged**